abstractions/dotnet/src/authentication/IAccessTokenProvider.cs

```diff
@@ -0,0 +1,20 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Kiota.Abstractions.Authentication;
+
+/// <summary>
+/// Defines a contract for obtaining access tokens for a given url.
+/// </summary>
+public interface IAccessTokenProvider
+{
+    /// <summary>
+    ///     This method is called by the <see cref="BaseBearerTokenAuthenticationProvider" /> class to get the access token.
+    /// </summary>
+    /// <param name="uri">The target URI to get an access token for.</param>
+    /// <returns>A Task that holds the access token to use for the request.</returns>
+    Task<string> GetAuthorizationTokenAsync(Uri uri);
```

Comment:
For dotnet, I'm realizing we might want to add an extra default `CancellationToken = default` parameter from the `IAuthenticationProvider's` `AuthenticateRequestAsync` method to the `IAccessTokenProvider` incase the `GetAuthorizationToken` is cancelled.
I'm happy to sort this out in a separate PR/work item if you agree.

Reply:
I agree, and please go ahead and directly push your commit to this branch.

Reply:
https://github.com/microsoft/kiota/pull/1050/commits/0c1e85acb6e394b9ebe598daa5e133467d5f2175